### PR TITLE
Custom select updates

### LINKED ISSUE #20316: .c-select-sm and .c-select-lg missing
If you have `.c-select form-control form-control-lg` it does not use proper padding and so it is not a `lg`, it's tiny.

### LINKED ISSUE #23823: Add a mention and/or example of .custom-select-sm to Custom Select section
This would be a trivial update, but: can the [Select Menu](https://getbootstrap.com/docs/4.0/components/forms/#select-menu) section of the custom form controls be updated to include a mention (or example) showing that there is a sizing class for a smaller version? My last attempt to issue a pull-request against the docs directory was auto-closed, or I'd just add it myself...

I could not find a mention of `.custom-select-sm` anywhere on that page, so I had to check the bootstrap.css file to find that there was, in fact, a sizing option.

**On a related tangent:** is there a reason not to also have a `-lg` version, or am I simply missing something?

### LINKED ISSUE #24504: Custom-select doesn't support single select with size attribute set above 1
- Bootstrap V4.0.0.-beta.2
- Operating system and version (any)
- Browser and version All
- Reduced test case https://codepen.io/tmorehouse/pen/boJQLL

The `size` attribute controls the number of visible "rows" the select displays. It is commonly used in conjunction with the `multiple` attribute, but can be used on single selects to produce a "list-box" style select (except on some mobile browsers).

When used on a `.custom-select`, not in `multiple` mode, the `size` attribute is not respected and only  single option (and bits of the next option) are shown (as well as bits of the custom select indicator):

image

As an aside note, when in multiple mode, there is also excess padding on the right (where the custom indicator would be on a single select)

Workaround CSS:
```css
.custom-select[multiple],
.custom-select[size]:not([size="1"])
{
    height: auto;
    background-image: none;
    /* Fix for excess right padding */
    padding-right: 0.75rem;
}
```

### LINKED ISSUE #24597: Custom Select Border and Custom File Browser when active or focus not displaying properly
only the normal 
the validation variant works

testet on FF and Chrome win 10

unbenannt-1

### PR DESCRIPTION
**Pull request by mdo**
Few fixes in one PR coming here:

- Fixes #24504: Add support for size attribute on custom selects.
- Fixes #20316 and fixes #23823:  Add large custom select, document it and the small variant.
- Fixes #24597 where custom file inputs had an inconsistent focus shadow.

Also need to look into #23058.

### CODE REVIEW

**Review by andresgalante — APPROVED** [review]

File: scss/_custom-forms.scss [hunk 1]
```diff
@@ -149,7 +149,7 @@
   &:focus {
     border-color: $custom-select-focus-border-color;
     outline: 0;
-    @include box-shadow($custom-select-focus-box-shadow);
+    box-shadow: $custom-select-focus-box-shadow;
```
Comment (XhmikosR) [inline scss/_custom-forms.scss]: Is this intentional?

File: scss/_custom-forms.scss [hunk 2]
```diff
@@ -162,9 +162,12 @@
     }
   }
 
-  &[multiple] {
+  &[multiple],
+  &[size]:not([size="1"]) {
     height: auto;
+    padding-right: $custom-select-padding-x;
     background-image: none;
+
```
Comment (XhmikosR) [inline scss/_custom-forms.scss]: This doesn't seem intentional :P

File: scss/_custom-forms.scss (hunk 1 above)
Comment (mdo) [reply scss/_custom-forms.scss]: Yes, we want the focus state to always be there regardless of the global shadow variable. This matches our other inputs.

**Comment by mdo** [comment]
Going to get this merged in a few and then tackle #23058.

**Pull request merged**